### PR TITLE
Introduce get|set_layout helpers

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Introduce `ov::layout::get|set_layout()` helpers

### CODE REVIEW

File: src/bindings/python/src/pyopenvino/graph/layout_helpers.cpp [hunk 1]
```diff
@@ -23,4 +23,6 @@ void regmodule_graph_layout_helpers(py::module m) {
     mod.def("height_idx", &ov::layout::height_idx, py::arg("layout"));
     mod.def("has_width", &ov::layout::has_width, py::arg("layout"));
     mod.def("width_idx", &ov::layout::width_idx, py::arg("layout"));
+    mod.def("get_layout", (ov::Layout(*)(const ov::Output<const ov::Node>&)) & ov::layout::get_layout, py::arg("port"));
```
Comment (nosovmik) [inline src/bindings/python/src/pyopenvino/graph/layout_helpers.cpp:26]: would be good to add python tests as well

File: src/core/src/layout.cpp [hunk 2]
```diff
@@ -469,4 +469,27 @@ std::string LayoutAttribute::to_string() const {
...
 
+ov::Layout get_layout(const ov::Output<const ov::Node>& output) {
+    auto it = output.get_rt_info().find(ov::LayoutAttribute::get_type_info_static());
+    if (it == output.get_rt_info().end()) {
+        return {};
+    }
+    return it->second.as<ov::LayoutAttribute>().value;
+}
+
+ov::Layout get_layout(const ov::Output<ov::Node>& output) {
```
Comment (nosovmik) [inline src/core/src/layout.cpp:480]: ov::layout::get_layout

File: src/core/src/layout.cpp [hunk 3]
```diff
@@ -469,4 +469,27 @@ std::string LayoutAttribute::to_string() const {
...
+    return get_layout(ov::Output<const ov::Node>(output.get_node(), output.get_index()));
+}
+
+void set_layout(ov::Output<ov::Node> output, const ov::Layout& layout) {
```
Comment (nosovmik) [inline src/core/src/layout.cpp:484]: ov::layout::set_layout

File: src/bindings/python/src/pyopenvino/graph/layout_helpers.cpp [hunk 4]
```diff
@@ -23,4 +23,6 @@ void regmodule_graph_layout_helpers(py::module m) {
...
+    mod.def("set_layout", &ov::layout::set_layout, py::arg("port"), py::arg("layout"));
```
Comment (ilya-lavrenov) [inline src/bindings/python/src/pyopenvino/graph/layout_helpers.cpp:27]: please, add python tests as well :)

**Review by ilya-lavrenov — APPROVED** [review]

File: src/core/tests/preprocess.cpp [hunk 5]
```diff
@@ -194,6 +194,7 @@ TEST(pre_post_process, tensor_element_type_and_scale) {
     EXPECT_EQ(f->get_parameters().front()->get_element_type(), element::f32);
     EXPECT_EQ(f->get_output_element_type(0), element::i8);
     EXPECT_EQ(f->get_parameters().front()->get_layout(), Layout());
+    EXPECT_EQ(ov::layout::get_layout(f->get_parameters().front()), Layout());
```
Comment (nosovmik) [inline src/core/tests/preprocess.cpp:197]: Suggest to add tests for 'set_layout' direct call and also for error cases (when Output<Node> is not related to Parameter or Result)

File: src/core/tests/preprocess.cpp [hunk 6]
```diff
@@ -208,6 +209,7 @@ TEST(pre_post_process, convert_color_nv12_rgb_single) {
     EXPECT_EQ(f->get_parameters().size(), 1);
     EXPECT_EQ(f->get_parameters().front()->get_element_type(), element::u8);
     EXPECT_EQ(f->get_parameters().front()->get_layout(), "NHWC");
+    EXPECT_EQ(ov::layout::get_layout(f->get_parameters().front()), "NCHW");
```
Comment (nosovmik) [inline src/core/tests/preprocess.cpp:212]: can it be just `ov::layout::get_layout(f->input(0))`?

File: src/bindings/python/src/pyopenvino/graph/layout_helpers.cpp (hunk 4 above)
Comment (ilyachur) [reply src/bindings/python/src/pyopenvino/graph/layout_helpers.cpp:27]: @AlexeyLebedev1 Could you provide the right place for python tests?

Comment (ilyachur) [reply src/bindings/python/src/pyopenvino/graph/layout_helpers.cpp:27]: Tests were added

**Review by ilya-lavrenov — APPROVED** [review]

File: src/bindings/python/tests/test_ngraph/test_basic.py [hunk 7]
```diff
@@ -624,6 +624,23 @@ def test_coordinate_diff_iteration_methods():
     assert np.equal(coordinateDiff, data2).all()
 
 
+def test_get_and_set_layout():
+    shape = [2, 2]
+    parameter_a = ops.parameter(shape, dtype=np.float32, name="A")
+    parameter_b = ops.parameter(shape, dtype=np.float32, name="B")
+
+    model = parameter_a + parameter_b
```
Comment (AlexeyLebedev1) [inline src/bindings/python/tests/test_ngraph/test_basic.py:632]:
```suggestion
    model = Model(parameter_a + parameter_b, [parameter_a , parameter_b])
```

**Review by AlexeyLebedev1 — APPROVED** [review]

File: src/core/include/openvino/core/layout.hpp [hunk 8]
```diff
@@ -129,6 +130,21 @@ OPENVINO_API bool has_width(const Layout& layout);
 ///
 OPENVINO_API std::int64_t width_idx(const Layout& layout);
 
+/// \brief Sets Layout of port
+///
+/// \throws ov::Exception if port is not connected with Result or Parameter
+OPENVINO_API void set_layout(ov::Output<ov::Node> output, const ov::Layout& layout);
```
Comment (ilya-lavrenov) [inline src/core/include/openvino/core/layout.hpp:136]: what if I `set_layout` to the node before the `Result`? I suppose it should work

**Pull request merged**